client/api/src/main/java/org/eclipse/microprofile/graphql/client/Response.java

```diff
@@ -18,18 +18,46 @@
 
 import javax.json.JsonObject;
 import java.util.List;
+import java.util.Map;
 
 public interface Response {
 
+    /**
+     * The 'data' object contained in the response.
+     * Can be JsonValue.NULL if the response contains an empty field, or `null` if the response
+     * does not contain this field at all.
+     */
     JsonObject getData();
 
-    List<Error> getErrors();
+    /**
+     * List of errors contained in this response.
+     */
+    List<GraphQLError> getErrors();
 
+    /**
+     * Transform the contents of the `rootField` from this response into a list of objects
+     * of the requested type.
+     */
     <T> List<T> getList(Class<T> dataType, String rootField);
 
+    /**
+     * Transform the contents of the `rootField` from this response into an object
+     * of the requested type.
+     */
     <T> T getObject(Class<T> dataType, String rootField);
 
+    /**
+     * If this response contains any data, this returns `true`; `false` otherwise.
+     */
     boolean hasData();
 
+    /**
+     * If this response contains at least one error, this returns `true`; `false` otherwise.
+     */
     boolean hasError();
+
+    /**
+     * Get transport-specific metadata that came from the server with this response.
+     */
+    List<Map.Entry<String, String>> getTransportMeta();
```

Comment:
This seems like a strange choice of data structures - why not just a `Map<String, String>`?

Reply:
Headers are multi-maps, i.e. every key can have multiple values. The alternative representation would be `Map<String, List<String>>`, which I prefer, as it's easier to handle and the intent is clearer. IIRC, @jmartisk had asked exactly this question somewhere.

Reply:
I don't remember it very well, but I think one reason for choosing `Map.Entry<String, String>` was that it makes it possible to preserve the exact order of the header lines in the HTTP response, which is not possible with `Map<String, List<String>>` in case that headers with the same key were not grouped together in the response.
Whether that really matters for GraphQL clients, I don't think so. Maybe there were some other reasons beyond that, but I don't know. I'm tending to think that `Map<String, List<String>>` would indeed be better

Reply:
I changed it to `Map<String, List<String>>`

Reply:
One more reason was probably that it was easier to work with `Map.Entry<String, String>` because Vert.x `MultiMap` has a `Map.Entry<String, String> getEntries()` method. But that's not a relevant reason to use that type in MicroProfile API.